File: C/pointers_arrays.c
#include <stdio.h>

// Function to manipulate an array using pointers
void manipulate_array(int* arr, int size) {
    for (int i = 0; i < size; i++) {
        arr[i] *= 2;
    }
}

int main() {
    int arr[] = {1, 2, 3, 4, 5};
    int size = sizeof(arr) / sizeof(arr[0]);
    manipulate_array(arr, size);

    printf("Array after manipulation: ");
    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    return 0;
}

File: Java/PointersArrays.java
// pointers in Java are not like C/C++ pointers. In Java, pointers are used to store the memory address of an object.
public class PointersArrays {

    public static void manipulateArray(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            arr[i] *= 4;
        }
    }

    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5};
        manipulateArray(arr);

        System.out.print("Array after manipulation: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}
